src/wesnoth.cpp

```diff
@@ -844,6 +847,9 @@ static int do_gameloop(const std::vector<std::string>& args)
 			break;
 		case gui2::dialogs::title_screen::REDRAW_BACKGROUND:
 			break;
+		case gui2::dialogs::title_screen::RELOAD_UI:
+			gui2::init(prefs::get().gui_theme(), false);
```

Comment:
IMO it'd be better to leave `init` as is and add a new method to set the current theme and activate it. `init` could perhaps use that method as well.

Reply:
Please see 3795673106744245e1f5e9fc3b0b806df6a75ef7. Does that work?

Reply:
Yeah, that looks good to me. I'd at least remove the `force_read` parameter from init again now. IMO `current_theme` is not that useful either if each caller just gets it from prefs anyway.

Reply:
Can `if (guis_cfg.empty() || force_read) {` check be removed since we have `switch_theme` now?

> IMO current_theme is not that useful either if each caller just gets it from prefs anyway.

Needed for #9057. I'll remove it later if it becomes unnecessary. Without that, if an addons switches themes, it will affect the global preferences.

Reply:
I don't see how #9057 is related yet but if you have plans to make use of it there then ok.

Reply:
After some thinking, I've removed the arguments in https://github.com/wesnoth/wesnoth/pull/9117/commits/506f0be4748bc078f2f48adc49ceb9a7ef6d2e50. However, `init()` is now responsible for only initializing the themes. It won't switch automatically. I think this makes it more obvious what's going on, and also allows me to remove the `preferences.hpp` dependency. I didn't like the fact that `init()` used the preferences under the hood.